src/CurlFormatter.php

```diff
@@ -42,7 +42,7 @@ public function format(Request $request)
                 $content = http_build_query((array)$data, '', '&', $this->encodingType);
                 $request->setFullUrl(null);
                 $url = $request->getFullUrl();
-                $url .= (strpos($url, '?') === false) ? '?' : '&';
+                $url .= ($url === null || strpos($url, '?') === false) ? '?' : '&';
```

Comment:
I don't think that in any valid scenario `$url` can be `null` here.

Reply:
The tests will fail with a "strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated" error without it.
And also according PhpStorm it can:
![image](https://github.com/yiisoft/yii2-httpclient/assets/1292337/020214c3-692d-472e-83e5-211b7a39a479)

Reply:
There were 2 errors:

1) yiiunit\extensions\httpclient\CurlFormatterTest::testFormatMethodGet
strpos(): Passing null to parameter # 1 ($haystack) of type string is deprecated

/me/yii2-httpclient/src/CurlFormatter.php:45
/me/yii2-httpclient/tests/CurlFormatterTest.php:52
/me/yii2-httpclient/vendor/phpunit/phpunit/phpunit:52
/me/yii2-httpclient/vendor/bin/phpunit:123

2) yiiunit\extensions\httpclient\UrlEncodedFormatterTest::testFormatMethodGet
strpos(): Passing null to parameter # 1 ($haystack) of type string is deprecated

/me/yii2-httpclient/src/UrlEncodedFormatter.php:51
/me/yii2-httpclient/tests/UrlEncodedFormatterTest.php:47
/me/yii2-httpclient/vendor/phpunit/phpunit/phpunit:52
/me/yii2-httpclient/vendor/bin/phpunit:123

Reply:
The `Request::getUrl()` can return `null` when `Request::$_url` is not set, so the PhpDocs for `Request::getUrl()` (`@return string|array`) is incorrect. If we want to ensure only those types we should throw an Exception when `Request::$_url` is not set.

Reply:
I would definitely fix these tests by setting url for request before calling `format()`. Current test looks like a bad usage for me and there should be some kind of error in that case.

Reply:
@samdark Sure. But what to with the PhpDocs for `Request::getUrl()` since that return type (`string|array`) is incorrect?